mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,19 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+
+        // check if indices is one-based and in ascending order
```

Comment:
There seems to be some performance difference.
```
    val indices = 1 to 200000000
    var start = System.nanoTime()

    var previous = -1
    var i = 0
    val indicesLength = indices.size
    while (i < indicesLength) {
      if (indices(i) <= previous) {
        throw new IllegalArgumentException("indices should be one-based and in ascending order")
      }
      previous = indices(i)
      i += 1
    }

    println("while: " + (System.nanoTime() - start).toDouble / 1e9)
    start = System.nanoTime()

    val g = indices.sliding(2).forall(p => p(0) < p(1))
    println("sliding: " + (System.nanoTime() - start).toDouble / 1e9)
```
while: 0.088418226
sliding: 37.128311352

I also used jconsole to collect the memory usage. The "sliding" way consumes significantly higher memory and triggers GC frequently/